www/components/FetchPackage.tsx

```diff
@@ -0,0 +1,32 @@
+import { Box } from "@welcome-ui/box";
+import React from "react";
+import useSWR from "swr";
+
+import * as docs from "@/lib/docs";
+import * as url from "@/lib/url";
+
+const fetcher = (input: RequestInfo | URL, init?: RequestInit) =>
+  fetch(input, init).then((res) => res.json());
+
+interface Props {
+  name?: string;
+  content: (pkg: docs.Pkg) => React.ReactElement;
+}
+
+export default function FetchPackage({ name, content }: Props) {
+  if (name === undefined) {
+    return null;
+  }
+  const pkgJsonUrl = url.pkgJson(name);
+  const { data, error } = useSWR<docs.Pkg, Error>(pkgJsonUrl, fetcher);
+
+  if (error)
+    return (
+      <Box>
+        Failed to load <code>{pkgJsonUrl}</code>: {error.message}
+      </Box>
+    );
+  if (!data) return <Box>Loading...</Box>;
```

Comment:
not in this PR, but maybe https://www.welcome-ui.com/components/loader

Reply:
👍  https://github.com/zsol/py.wtf/issues/26